coderd/database/dbfake/dbfake.go

```diff
@@ -0,0 +1,124 @@
+package dbfake
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbauthz"
+	"github.com/coder/coder/v2/coderd/database/dbgen"
+	"github.com/coder/coder/v2/coderd/database/dbtime"
+	"github.com/coder/coder/v2/coderd/provisionerdserver"
+	"github.com/coder/coder/v2/coderd/telemetry"
+	"github.com/coder/coder/v2/provisionersdk/proto"
+	sdkproto "github.com/coder/coder/v2/provisionersdk/proto"
+)
+
+// CreateWorkspace inserts a workspace into the database.
+func CreateWorkspace(t testing.TB, db database.Store, seed database.Workspace) database.Workspace {
```

Comment:
The thing to maybe comment is that this does not create the full dependency tree from our application logic POV.

That meaning the template version, files, workspace build, provisioner job. I imagine this just the bare minimum to pass the db constraints?

Should we name this `CreateMinimumWorkspace`? I would be ok if we just comment the functionality too.

Reply:
It's actually commented below where we insert the template and logic n such if those aren't provided